### ISSUE TITLE
rdbms specific aggregation functions 

### ISSUE THREAD

**Issue by mobiworx** [slick/slick#923]
I know of the possibility to use functions defined in my rdbms in slick by using "SimpleFunction" variants. 

Currently I'm in need to use postgres's "string_agg" aggregation function ("group concat") but I cannot find a way to use it. I have this code:

``` scala
q.groupBy(_._1.id).map {
    case (id, group) => (id, group.map(_._1.objectname).max, group.map(_._2._2).min)
}
```

Instead of "group.map(_._2._2).min" I want to use basically "group.map(_._2._2).string_agg(',')". If that is not possible I think it should explicitly be documented in the "aggregating" section of slick. I'm happy to supply a PR for the docs if that's the case.

**Comment by szeiger** [comment]
Yes, I think group_concat would be a welcome addition.

**Comment by TimothyKlim** [comment]
Is there any news? It's really painful to use slick without db specific aggregation methods like "array_agg" in postgres.

**Comment by deanchen** [comment]
+1

**Comment by mobiworx** [comment]
The option I found helpful is to encapsulate the usage of "array_agg" in a database view and just use this view with slick. This way you do not need "array_agg" directly in slick.

**Comment by TimothyKlim** [comment]
@mobiworx this is a solution if you have one or two queries with db specific aggregation methods but for project with complex structure you cannot create new view for each query every time.

**Comment by mobiworx** [comment]
I agree with you completely. But as of now that is the only option we have besides writing SQL directly. I made my views in a way that let me reuse them in joins instead of really having to create a new at each new query.

**Comment by evbo** [comment]
Note that there is an extension library for supporting aggregate functions but it's only for postgres: https://github.com/tminglei/slick-pg

**Comment by Mosho1** [comment]
any update on this? I would really like to use group_concat in mysql, willing to implement myself but I want to at least know it would work...
